### PR TITLE
Handle empty tuple struct def with comment properly

### LINKED ISSUE #919: Handle empty struct defs properly
We either don't reformat (struct structs) or lose comments (tuple structs). We should do neither.

e.g.,

```
struct Foo( /* a comment */ );
struct Foo();
struct Foo{ /* a comment */ }
struct Foo{}
```

should be idempotent

### PR DESCRIPTION
**Pull request by topecongiro**
This PR closes #919.

### CODE REVIEW

File: tests/source/structs.rs [hunk 1]
```diff
@@ -166,7 +166,7 @@ struct Foo {
 
     }
 struct Foo { /* comment */ }
-struct Foo();
+struct Foo( /* comment */ );
```
Comment (nrc) [inline tests/source/structs.rs]: The new line should be in addition to the old one, not replace it.

**Comment by topecongiro** [comment]
Fixed a nit in the test.

**Comment by nrc** [comment]
Thanks for the fix, needs a rebase now, sorry.

**Comment by topecongiro** [comment]
Rebased.

**Pull request merged**